index/scorch/persister.go

```diff
@@ -37,8 +37,26 @@ import (
 
 var DefaultChunkFactor uint32 = 1024
 
-var DefaultPersisterNapTimeMSec int = 2000 // ms
-
+// DefaultPersisterNapTimeMSec is kept to zero as this helps in direct
+// persistence of segments with the default safe batch option.
+// If the default safe batch option results in high number of
+// files on disk, then users may initialise this configuration parameter
+// with higher values so that the persister will nap a bit within it's
+// work loop to favour better in-memory merging of segments to result
+// in fewer segment files on disk. But that may come with an indexing
+// performance overhead.
+// Unsafe batch users are advised to override this to higher value
+// for better performance especially with high data density.
+var DefaultPersisterNapTimeMSec int = 0 // ms
+
+// DefaultPersisterNapUnderNumFiles helps in controlling the pace of
+// persister. At times of a slow merger progress with heavy file merging
+// operations, its better to pace down the persister for letting the merger
+// to catch up within a range defined by this parameter.
+// Fewer files on disk (as per the merge plan) would result in keeping the
+// file handle usage under limit, faster disk merger and a healthier index.
+// Its been observed that such a loosely sync'ed introducer-persister-merger
+// trio results in better overall performance.
 var DefaultPersisterNapUnderNumFiles int = 1000
```

Comment:
(without checking the code) do we not need to reset this to a default of zero?

Reply:
Some form of persister-merger sync is needed, else we end up having too many files on disk..and his become very severe with smaller batches. 
Please let me know if something needs to be added/updated in the comments above. I thought that comments already explain why we left a non zero default here..

Reply:
Ok let me rephrase my question - did you test these settings to check if the regression went away? :)

Reply:
indeed! for beer-search app.

Reply:
```
./beer-search 
2018/11/08 12:33:35 GOMAXPROCS: 8
2018/11/08 12:33:35 Creating new index...
2018/11/08 12:33:35 
 parsePersisterOptions {PersisterNapTimeMSec:0 PersisterNapUnderNumFiles:1000}
2018/11/08 12:33:35 Listening on :8094
2018/11/08 12:33:35 Indexing...
2018/11/08 12:33:35 Indexed 1000 documents, in 0.33s (average 0.33ms/doc)
2018/11/08 12:33:35 Indexed 2000 documents, in 0.64s (average 0.32ms/doc)
2018/11/08 12:33:36 Indexed 3000 documents, in 0.94s (average 0.31ms/doc)
2018/11/08 12:33:36 Indexed 4000 documents, in 1.24s (average 0.31ms/doc)
2018/11/08 12:33:36 Indexed 5000 documents, in 1.55s (average 0.31ms/doc)
2018/11/08 12:33:37 Indexed 6000 documents, in 1.86s (average 0.31ms/doc)
2018/11/08 12:33:37 Indexed 7000 documents, in 2.19s (average 0.31ms/doc)
2018/11/08 12:33:37 Indexed 7303 documents, in 2.29s (average 0.31ms/doc)
```